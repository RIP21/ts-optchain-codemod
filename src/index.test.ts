import pluginTester from "babel-plugin-tester";
import transform from "./index";
const { format } = require("prettier");

const buildTestCase = ({ title, code }) => {
  return {
    title,
    code: `
      import oc from 'ts-optchain'
      ${code}`
  };
};

pluginTester({
  pluginName: "ts-optchain-codemod",
  plugin: transform,
  snapshot: true,
  formatResult: output =>
    format(output, {
      semi: true,
      singleQuote: true,
      parser: "babel"
    }),
  tests: [
    buildTestCase({
      title: "Simple test",
      code: `oc(one).two('default')`
    })
  ]
});
