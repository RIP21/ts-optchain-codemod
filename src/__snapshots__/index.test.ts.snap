// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-optchain-codemod Simple test: Simple test 1`] = `

import oc from 'ts-optchain'
oc(one).two('default')

      ↓ ↓ ↓ ↓ ↓ ↓
one?.two?.('default');
"
`;
